modules/timer/timer.cpp

```diff
@@ -0,0 +1,327 @@
+static void decrementSecondDigitValue   (int& fourthDigitInitialValue, int& thirdDigitInitialValue, int& secondDigitInitialValue, int& firstDigitInitialValue);
+static void decrementThirdDigitValue    (int& fourthDigitInitialValue, int& thirdDigitInitialValue, int& secondDigitInitialValue, int& firstDigitInitialValue);
+static void decrementFourthDigitValue   (int& fourthDigitInitialValue, int& thirdDigitInitialValue, int& secondDigitInitialValue, int& firstDigitInitialValue);
+
+static int * readInitialValueOfSpecificMode             (modes_t mode);
+static int * readTemporalValueOfSpecificMode            (modes_t mode);
+static void  copyInitialToTemporalValuesOfTheActiveMode ();
+static void  assignValuesForANewCicle                   ();
+
+//=====[Implementations of public functions]===================================
+void checkValuesOfCurrentModeAndReassignIfCorresponds() {
+    if (currentMode == MODE_EXERCISE) {
+        if (
+            exerciseModeTemporalValues[3]   == 0  && 
+            exerciseModeTemporalValues[2]   == 0  && 
+            exerciseModeTemporalValues[1]   == 0  && 
+            exerciseModeTemporalValues[0]   == 0
+        ) {
+            currentMode = MODE_REST;
+            return;
+        } else {
+            return;
+        }
+    }
+
+    if (currentMode == MODE_REST) {
+        if (
+            restModeTemporalValues[3]   == 0  && 
+            restModeTemporalValues[2]   == 0  && 
+            restModeTemporalValues[1]   == 0  && 
+            restModeTemporalValues[0]   == 0
+        ) {
+            currentMode = MODE_REPETITIONS;
+            return;
+        } else {
+            return;
+        }
+    }
+
+    if (currentMode == MODE_REPETITIONS) {
+        if (
+            repetitionModeTemporalValues[3]   == 0  && 
+            repetitionModeTemporalValues[2]   == 0  && 
+            repetitionModeTemporalValues[1]   == 0  && 
+            repetitionModeTemporalValues[0]   == 0
+        ) {
+            setCountdownState(false);
+            informThatCountdownIsFinished();
+            currentMode = MODE_EXERCISE;
+            return;
+        } else {
+            assignValuesForANewCicle();
+            return;
+        }
+    }
+    return;
+}
+
+int * readInitialValueOfTheModeActive() {
+    switch (currentMode){
+        case MODE_EXERCISE:     return exerciseModeInitialValues;
+        case MODE_REST:         return restModeInitialValues;
+        case MODE_REPETITIONS:  return repetitionModeInitialValues;
+        default:                return errorValues;
+    }
+}
+
+int * readTemporalValueOfTheModeActive() {
+    switch (currentMode){
+        case MODE_EXERCISE:     return exerciseModeTemporalValues;
+        case MODE_REST:         return restModeTemporalValues;
+        case MODE_REPETITIONS:  return repetitionModeTemporalValues;
+        default:                return errorValues;
+    }
+}
+
+modes_t readModeActive() {
+    return currentMode;
+}
+
+void updateMode() {
```

Comment:
Me parece que sería más claro si acá indicas al menos mediante un comentario dónde y cómo es que se cambia el modo; lo mismo en todos los casos similares a este, así aumenta la legibilidad del código.

Reply:
El modo se cambia de dos maneras:

- Cuando se presiona el botón de selección, llamando a la función `updateMode()`
- Cuando termina una cuenta regresiva (por ej tiempo de ejercicio) y hay que descontar una repetición e iniciar la cuenta regresiva del otro modo (por ej tiempo de descanso). En este caso se llama a la función `checkValuesOfCurrentModeAndReassignIfCorresponds()`.

En el primer caso, el usuario interviene para que el programa llame a esa función. En el segundo caso se llama de forma automática si corresponde. 

Si entiendo bien la observación, sería poner un comentario en la función `updateMode()` mencionando que el modo se cambia cuando el usuario presiona el botón de selección. Está bien así?

Reply:
Sí, correcto. Sería poner un comentario en la función updateMode() mencionando que el modo se cambia cuando el usuario presiona el botón de selección. Doy por resuelto este comentario mío.

Reply:
Irresolví la conversación porque me olvidé de incluir este cambio. En el commit ['add comments to led and timer modules'](https://github.com/ferauzalez/ARMBookCurso2024_TP1/pull/3/commits/9a832068cde91f1d3f34c751d1d5ac6ccc3f79cc) agrego el cambio. No sé como es el procedimiento para notificarte específicamente esto. Por favor marcá como resuelto nuevamente cuando lo veas.

Reply:
Irresolví la conversación porque me olvidé de incluir este cambio. En el commit ['add comments to led and timer modules'](https://github.com/ferauzalez/ARMBookCurso2024_TP1/pull/3/commits/9a832068cde91f1d3f34c751d1d5ac6ccc3f79cc) agrego el cambio. No sé como es el procedimiento para notificarte específicamente esto. Por favor marcá como resuelto nuevamente cuando lo veas.

Reply:
Ok, lo doy por resuelto.